src/main/java/org/opensearch/knn/index/mapper/CompressionLevel.java

```diff
@@ -124,12 +136,22 @@ public RescoreContext getDefaultRescoreContext(Mode mode, int dimension, Version
             }
             return defaultRescoreContext;
         }
-        return null;
-    }
 
-    @VisibleForTesting
-    RescoreContext getDefaultRescoreContext(Mode mode, int dimension) {
-        return getDefaultRescoreContext(mode, dimension, Version.CURRENT);
+        // Special handling for Lucene BBQ (x32 compression)
+        if (this == x32 && engine == KNNEngine.LUCENE && version.onOrAfter(Version.V_3_2_0)) {
```

Comment:
Should be after `V_3_3_0`, since this was not added as a part of 3.2.

Reply:
no, version.onOrAfter(Version.V_3_3_0)